### PR TITLE
Refactor f8_t, add bf8_t

### PR DESCRIPTION
**Pull request by geyyer**
- Add bf8_t type
- Use BitInt for f8_t and bf8_t
- Add bf8_t conversion methods
- Add tests for both f8_t and bf8_t

### CODE REVIEW

File: include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp [hunk 1]
```diff
@@ -89,6 +89,7 @@ struct PassThrough
     }
 #endif
 
+#if defined CK_ENABLE_FP8 || defined CK_ENABLE_BF8
```
Comment (aosewski) [inline include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp]: Do you need `CK_ENABLE_BF8` here? I don't see bf8 used.

File: include/ck/utility/data_type.hpp [hunk 2]
```diff
@@ -12,7 +12,10 @@ using half_t  = _Float16;
 #ifdef CK_EXPERIMENTAL_BIT_INT_EXTENSION_INT4
 using int4_t = _BitInt(4);
 #endif
-using f8_t = uint8_t;
+#if defined CK_ENABLE_FP8 || defined CK_ENABLE_BF8
```
Comment (aosewski) [inline include/ck/utility/data_type.hpp]: Just wonder why we have to tightly couple those two data types?

File: library/include/ck/library/tensor_operation_instance/device_operation_instance_factory.hpp [hunk 3]
```diff
@@ -17,10 +17,12 @@ namespace instance {
 using F64  = double;
 using F32  = float;
 using F16  = ck::half_t;
-using F8   = ck::f8_t;
 using BF16 = ck::bhalf_t;
 using I8   = int8_t;
 using I32  = int32_t;
+#if defined CK_ENABLE_FP8 || defined CK_ENABLE_BF8
+using F8 = ck::f8_t;
```
Comment (aosewski) [inline library/include/ck/library/tensor_operation_instance/device_operation_instance_factory.hpp:24]: Why you don't define an alias for `bf8_t` ?

**Review by aosewski — COMMENTED** [review]
In general look very good! Great job! 

I just wonder whether we need to tightly couple those two types `f8_t` and `bf8_t` ? I'm asking since you always enable **both** of them whenever **any** of them is enabled via macros.

**Comment by zjing14** [comment]
@geyyer Why we need `#if defined CK_ENABLE_FP8 || defined CK_ENABLE_BF8` everywhere?

File: include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp (hunk 1 above)
Comment (geyyer) [reply include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp]: Fixed in [2a51cf4](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/792/commits/2a51cf49c2918731bcb0e11a70fb77198cd12d56)

File: include/ck/utility/data_type.hpp (hunk 2 above)
Comment (geyyer) [reply include/ck/utility/data_type.hpp]: Fixed in [2a51cf4](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/792/commits/2a51cf49c2918731bcb0e11a70fb77198cd12d56)

File: library/include/ck/library/tensor_operation_instance/device_operation_instance_factory.hpp (hunk 3 above)
Comment (geyyer) [reply library/include/ck/library/tensor_operation_instance/device_operation_instance_factory.hpp:24]: Fixed in [2a51cf4](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/792/commits/2a51cf49c2918731bcb0e11a70fb77198cd12d56)

**Comment by geyyer** [comment]
> @geyyer Why we need `#if defined CK_ENABLE_FP8 || defined CK_ENABLE_BF8` everywhere?

@zjing14, I've decoupled fp8 and bf8 flags in [2a51cf4](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/792/commits/2a51cf49c2918731bcb0e11a70fb77198cd12d56), so we don't need it anymore.

File: include/ck/utility/f8_utils.hpp [hunk 4]
```diff
@@ -141,81 +126,70 @@ __host__ __device__ Y run_cast_to_f8(X x, uint32_t rng)
...
 {
     // check data type
-    constexpr bool is_half  = std::is_same<Y, half_t>::value;
-    constexpr bool is_float = std::is_same<Y, float>::value;
-    constexpr bool is_f8_t  = std::is_same<X, f8_t>::value;
-    constexpr bool is_bf8_t = std::is_same<X, bf8_t>::value;
+    // constexpr bool is_half  = std::is_same<Y, half_t>::value;
+    // constexpr bool is_float = std::is_same<Y, float>::value;
+    // constexpr bool is_f8_t  = std::is_same<X, f8_t>::value;
+    // constexpr bool is_bf8_t = std::is_same<X, bf8_t>::value;
```
Comment (aosewski) [inline include/ck/utility/f8_utils.hpp]: Leftovers.

**Review by aosewski — DISMISSED** [review]

File: include/ck/utility/f8_utils.hpp (hunk 4 above)
Comment (geyyer) [reply include/ck/utility/f8_utils.hpp]: Oops, thanks for noticing, removed in [6de7d10](https://github.com/ROCmSoftwarePlatform/composable_kernel/pull/792/commits/6de7d10dc96a679dd9b3d59d66a1201c9930092e)

**Review by aosewski — APPROVED** [review]

**Pull request merged**